sandbox/grist/predicate_formula.py

```diff
@@ -38,17 +39,17 @@ def parse_predicate_formula(formula):
   if isinstance(formula, six.binary_type):
     formula = formula.decode('utf8')
   try:
-    formula = replace_dollar_attrs(formula)
+    formula = get_dollar_replacer(formula).get_text()
     tree = ast.parse(formula, mode='eval')
     result = TreeConverter().visit(tree)
     for part in tokenize.generate_tokens(io.StringIO(formula).readline):
       if part[0] == tokenize.COMMENT and part[1].startswith('#'):
         result = ['Comment', result, part[1][1:].strip()]
         break
     return result
-  except SyntaxError as err:
+  except SyntaxError as e:
```

Comment:
I always catch errors under the name `e`, and for consistency shortened it on the spot when I saw it. Minor thing though?

Good point for the f-string.

Reply:
(F-string warning: the data engine still needs to be Python 2.7 compatible because of some legacy users, sorry! There can be exceptions, with some care, but right here seems a bad place for that exception.)

Reply:
Reverted the f-string change.